src/Forms/ChooseEncoding.cs

```diff
@@ -63,11 +63,12 @@ public void Initialize(string fileName)
             foreach (var enc in Configuration.AvailableEncodings)
             {
                 var item = new ListViewItem(new[] { enc.CodePage.ToString(), enc.WebName, enc.EncodingName });
-                listView1.Items.Add(item);
+                item.Tag = enc;

```

Comment:
store encoding in item tag for faster retrieve later on...

Reply:
Is actually slower in my tests...

Reply:
That is strange... Encoding.GetEncoding is actually calling `EncodingProvider.GetEncodingFromProvider(name)` which loops through a array of encoding... maybe there is some cache of something I will investigate more!

![image](https://user-images.githubusercontent.com/6579818/92611444-b438ab00-f2b0-11ea-951a-fc713227cab7.png)

![image](https://user-images.githubusercontent.com/6579818/92611036-442a2500-f2b0-11ea-9b4d-49e086231a3d.png)